draft-wang-ppm-dap-taskprov.md

```diff
@@ -56,69 +60,302 @@ collector). Thus it is up to each deployment of DAP to securely implement its
 own mechanism.
 
 This document describes a mechanism for configuring tasks that may be useful in
-many deployments.
+many deployments. The goal of this mechanism is to add transparency to the
```

Comment:
I continue to think that "transparency" is not the right sales pitch for this extension. The reason is this: To sell it this way, you'll need to (1) define transparency and (2) to make the case that vanilla DAP *isn't* transparent. So far I don't see a clear/useful definition of transparency for which this case can be made.

Suppose we define transparency this way: "A DAP deployment is transparent if what is being measured, and how, is evident to the client." Assuming this definition, it's not hard to argue that draft-wang-ppm-taskprov does not add any value relative to vanilla DAP. The important question is: What information does the client have now that it didn't have before? Quoting from the original issue (https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/issues/290):

> By sending all task parameters to clients then return to the server, we add transparency to the task that clients participate in Client has the choice to see what data has been collected with what parameters (like min_batch_size, max_batch_lifetime, or any differential privacy parameters if that's the privacy guarantee used).  For some clients these task parameters can even be hardcoded on the client side (for e.g. on a mobile device) to avoid any tempering from server.

Why is this not true for vanilla DAP? It seems like a deployment is free to surface this information to clients, regardless of how the task is configured. In fact, at some point the client software needs this information in order to participate in the protocol. It's also possible to hardcode DP parameters and use them for multiple tasks for vanilla DAP.

On the other hand, I think a *very* strong case for this extension can be made without talking about transparency at all. Namely, it specifies a means of configuring tasks without intervention of a third-party; all information necessary for aggregating a report is carried by the report itself. This is an elegant idea on its own.

Reply:
I agree with the part that transparency in the form of "client knows the parameters" is true to vanilla DAP. 

But what this extension adds is: the client's measurement will be aggregated using the parameters, because aggregators use these exact parameters to define a task. While in vanilla DAP, there's nothing stopping task authors from giving clients and helpers different parameters.

Reply:
That's definitely a useful property of this extension, and it should be called out in the intro. But I don't think it has to do with transparency.